cmd/server/server.go

```diff
@@ -133,7 +138,17 @@ func runServer(sa *serverArgs) error {
 	adapterMgr := adapterManager.NewManager(adapter.Inventory(), aspect.Inventory(), eval)
 	configManager := config.NewManager(eval, adapterMgr.AspectValidatorFinder(), adapterMgr.BuilderValidatorFinder(),
 		sa.globalConfigFile, sa.serviceConfigFile, time.Second*time.Duration(sa.configFetchIntervalSec))
-	handler := api.NewHandler(adapterMgr, adapterMgr.AspectMap())
+
+	var handler api.Handler
+	if sa.workerPoolSize == 0 {
+		handler = api.NewHandler(adapterMgr, adapterMgr.AspectMap())
+	} else {
+		poolsize := int(sa.workerPoolSize)
+		if poolsize < 0 {
```

Comment:
Rather, I should say I didn't know go failed compilation when invalid constant casts are performed.